### PR TITLE
Run test suites on subproject

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fix `make test` not running the crypto tests if a TLS test fails. Ensure that the reporting is correct.

## PR checklist

- [x] **changelog** not required because: maintainer facing only
- [x] **development PR** here
- [x] **framework PR** not required
- [x] **3.6 PR** not required because: bug introduced in development
- [x] **2.28 PR** not required because: bug introduced in development
- **tests**  provided + run `basic-build-test.sh`: https://mbedtls.trustedfirmware.org/job/mbedtls-restricted-release-new/33/

### CODE REVIEW

File: tests/scripts/run-test-suites.pl [hunk 1]
```diff
@@ -82,16 +83,18 @@ sub pad_print_center {
...
+    $suite =~ s!.*/!!;
     print "$suite ", "." x ( 72 - length($suite) - 2 - 4 ), " ";
     if( $suite =~ /$skip_re/o ) {
         print "SKIP\n";
         ++$suites_skipped;
         next;
     }
 
-    my $command = "$prefix$suite";
+    my $command = "$prefix$suite_path";
```
Comment (gilles-peskine-arm) [inline tests/scripts/run-test-suites.pl:97]: That only works if the auto-chdir behavior in `host_test.function` is enabled. Also it doesn't work on Windows with cmd. Patch incoming.

**Review by mpg — APPROVED** [review]
LGTM, thanks!

I also checked in the output of `basic-build-test.sh` that the number of tests reported hasn't changed compared to the latest nightly.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Comment by gilles-peskine-arm** [comment]
One DTLS test case failed [in the merge queue](https://mbedtls.trustedfirmware.org/job/mbed-tls-pr-head/job/gh-readonly-queue%2Fdevelopment%2Fpr-9558-1a09caa8a858ee1848fa30cb117ac2486d237020/1/). I'm going to try again to see if it's reproducible. If it isn't I'll file an issue.

**Pull request merged**